src/osd/PGBackend.h

```diff
@@ -555,17 +555,36 @@ typedef std::shared_ptr<const OSDMap> OSDMapRef;
      map<string, bufferlist> *out);
 
    virtual int objects_read_sync(
-     const hobject_t &hoid,
+     const hobject_t &soid,
+     const shard_id_t &shard,
      uint64_t off,
      uint64_t len,
      uint32_t op_flags,
-     bufferlist *bl) = 0;
-
+     unsigned size,
+     bufferlist *bl,
+     boost::optional<uint32_t> maybe_crc,
+     bool sparse) = 0;
+
+   struct ReadItem {
+     uint64_t off;
+     uint64_t len;
+     unsigned flags;
+     bufferlist *out;
```

Comment:
nit: `ceph::bufferlist` as, in headers, we shouldn't presume inclusion of a namespace.